h/api/models/elastic.py

```diff
@@ -117,7 +117,7 @@ def shared(self):
         elif len(read_perms) > 1 and self.groupid == '__world__' and gperm in read_perms:
             read_perms = [gperm]
 
-        if (read_perms != [gperm] and read_perms != [self.userid]):
+        if (gperm not in read_perms and read_perms != [self.userid]):
```

Comment:
It's worth noting that all the code in `h.api.models.elastic` is being used in the postgres migration script. So any change would need to be tested by running the migration again.